src/sql/parts/notebook/notebookInput.ts

```diff
@@ -8,150 +8,189 @@
+			let endLine = model.getLineCount();
+			let endCol = model.getLineLength(endLine);
+			this.textEditorModel.textEditorModel.applyEdits([{
+				range: new Range(1, 1, endLine, endCol),
+				text: content
+			}]);
 		}
-		return TPromise.wrap(true);
 	}
-}
 
+	isModelCreated(): boolean {
+		return this.getNotebookModel() !== undefined;
+	}
+
+	private getNotebookModel(): INotebookModel {
+		let editor = this.notebookService.listNotebookEditors().find(n => n.id === this.notebookUri.toString());
+		if (editor) {
+			return editor.model;
+		}
+		return undefined;
+	}
+
+	get onDidChangeDirty(): Event<void> {
+		return this._onDidChangeDirty.event;
+	}
+}
 
 export class NotebookInput extends EditorInput {
 	public static ID: string = 'workbench.editorinputs.notebookInput';
-
+	private _providerId: string;
+	private _providers: string[];
+	private _standardKernels: IStandardKernelWithProvider[];
+	private _connectionProfileId: string;
+	private _defaultKernel: azdata.nb.IKernelSpec;
+	private _isTrusted: boolean = false;
 	public hasBootstrapped = false;
 	// Holds the HTML content for the editor when the editor discards this input and loads another
 	private _parentContainer: HTMLElement;
 	private readonly _layoutChanged: Emitter<void> = this._register(new Emitter<void>());
+	private _model: NotebookEditorModel;
+	private _untitledEditorService: IUntitledEditorService;
+
 	constructor(private _title: string,
-		private _model: NotebookInputModel,
+		private resource: URI,
+		@ITextModelService private textModelService: ITextModelService,
+		@IUntitledEditorService untitledEditorService: IUntitledEditorService,
+		@IInstantiationService private instantiationService: IInstantiationService,
 		@INotebookService private notebookService: INotebookService,
-		@IDialogService private dialogService: IDialogService
 	) {
 		super();
-		this._model.onDidChangeDirty(() => this._onDidChangeDirty.fire());
+		this._untitledEditorService = untitledEditorService;
+		this.resource = resource;
+		this._standardKernels = [];
+		this.assignProviders();
 	}
 
 	public get notebookUri(): URI {
-		return this._model.notebookUri;
+		return this.resource;
+	}
+
+	public getName(): string {
+		if (!this._title) {
+			this._title = resources.basenameOrAuthority(this.resource);
+		}
+		return this._title;
 	}
 
 	public get providerId(): string {
-		return this._model.providerId;
+		return this._providerId;
 	}
 
-	public get providers(): string[] {
-		return this._model.providers;
+	public set providerId(value: string) {
+		this._providerId = value;
+	}
+
+	public get isTrusted(): boolean {
+		return this._isTrusted;
+	}
+
+	public set isTrusted(value: boolean) {
+		this._isTrusted = value;
 	}
 
 	public get connectionProfileId(): string {
-		return this._model.connectionProfileId;
+		return this._connectionProfileId;
 	}
 
 	public get standardKernels(): IStandardKernelWithProvider[] {
-		return this._model.standardKernels;
+		return this._standardKernels;
```

Comment:
why was all of this moved from notebookInputModel into notebookInput?

Reply:
Removed NotebookEditorModel' instantiation from two places as there is no significance in passing this object with NotebookInput. Instead, NotebookEditorModel is driven by NotebookInput.